app/models/validations.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+# validations used by multiple objects
+module Validations
+  # Validates obj.when.
+  # Includes setting it to current date if user set it to nil.
+  # used by: Observation, SpeciesList
+  def validate_when
+    self.when ||= Time.zone.now
```

Comment:
I believe this line is causing this to act effectively as a concern.  Any reference to "self" means you are adding on to a class which is what I am concerned about.  It makes it really hard to track down where things are happening.  Similarly, looking at this file it is really unclear what errors without looking at where this code is called from.  Instead, I would write this as something like:

def time_validation(when, errors)
  check_date(when, errors) && check_time(when, errors) && check_year(when, errors)
end

and then move the update of self.when back into the class where it belongs.

Reply:
@mo-nathan:
Thanks for taking the time to explain this. Sorry to be slow on the uptake.
I just pushed something that I think is fully responsive to your Comments.
It's less dry and wordier than a Concern: 
- It repeatedly passes arguments instead of relying on `self.where` 
- Adds a little duplication between `observation.rb` and `species_list.rb` for the same reason. 

But that looks to me like that's a necessary tradeoff for avoiding multiple inheritance. Is there a better way?